torch/distributed/rpc/api.py

```diff
@@ -137,13 +140,15 @@ def _wait_all_workers():
     if is_leader_worker:
         _on_leader_follower_report_shutdown_intent(self_worker_name)
     else:
-        rpc_sync(
-            leader_worker_name,
-            _on_leader_follower_report_shutdown_intent,
-            args=(self_worker_name,),
-        )
+        if exit_on_worker_done is None:
+            rpc_sync(
+                leader_worker_name,
+                _on_leader_follower_report_shutdown_intent,
+                args=(self_worker_name,),
+            )
 
-    _SHUTDOWN_PROCEED_SIGNAL.wait()
+    if not is_leader_worker or (is_leader_worker and exit_on_worker_done is None):
```

Comment:
Q: for this case `is_leader_worker and exit_on_worker_done is None`, who will be setting the event `_SHUTDOWN_PROCEED_SIGNAL`